diff --git a/node_modules/expo-modules-core/ios/Interfaces/ImageLoader/EXImageLoaderInterface.h b/node_modules/expo-modules-core/ios/Interfaces/ImageLoader/EXImageLoaderInterface.h
index 2b241d1..c8b2701 100644
--- a/node_modules/expo-modules-core/ios/Interfaces/ImageLoader/EXImageLoaderInterface.h
+++ b/node_modules/expo-modules-core/ios/Interfaces/ImageLoader/EXImageLoaderInterface.h
@@ -2,7 +2,11 @@
 
 #import <ExpoModulesCore/Platform.h>
 
+#if TARGET_OS_OSX
+typedef void (^EXImageLoaderCompletionBlock)(NSError *error, void *image);
+#else
 typedef void (^EXImageLoaderCompletionBlock)(NSError *error, UIImage *image);
+#endif
 
 @protocol EXImageLoaderInterface <NSObject>
 
